conftest.py

```diff
@@ -244,7 +266,12 @@ def get_test_launcher() -> str:
 
     @staticmethod
     def get_test_port() -> int:
-        return test_port
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
```

Comment:
Depending on when this gets merged, you might be able to swap to using [`smartsim._core.utils.network.find_free_port`](https://github.com/CrayLabs/SmartSim/blob/dragon_launcher/smartsim/_core/utils/network.py#L107-L125)

Reply:
Current plan is this PR will likely go in before the dragon work, so this fn is not yet defined, but we should consider using it as soon as the dragon work is merged!